clouddriver-core/src/test/java/com/netflix/spinnaker/clouddriver/security/AccountDefinitionSecretManagerTest.java

```diff
@@ -130,4 +138,41 @@ void canAccessAccountWhenUserAndAccountHaveNoPermissions() {
     assertThat(accountDefinitionSecretManager.canAccessAccountWithSecrets(username, accountName))
         .isTrue();
   }
+
+  @Test
+  void testVersionCompatibilityForGoogleSecretManager() throws Exception {
+
+    String credentials =
+        "{\n"
+            + "  \"type\": \"service_account\",\n"
+            + "  \"project_id\": \"my-test-project\",\n"
+            + "  \"private_key_id\": \"aaaaaaaaaaaaaaaaaaa\",\n"
+            + "  \"private_key\": \"-----BEGIN PRIVATE KEY-----\\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDmKDSt86RQ59Zx\\njezHgVF4IWCwbE6QzdnJGumPnpIvTS5/575A9grY5WB5s4H4FnrrmLVJe5T0mxc5\\nrJ4v9JKyHeTQn+OdMV0zwhJczIN+raROZ9GJyxgdYysiyRR1ajkd8aX+aiU5A4r9\\nFcIIcbLkrlltKfSV3I6tiE0oZja2sj+OE4+3b85NMiBUeusLH7GRglRDddnCAysY\\nvk2tIsYjI9m+f6r731rrmAU1SA5sx7pbRMJdWPVxCSrJy0F+AH/Hn1rTD+ga1m4+\\ngLuBbjZLkfFeg7xsqWKTImaMxKDNpiaNiUaBrBXS1u6IqfjDMakFU9tm5sbGuauG\\nXdIqYo0DAgMBAAECggEABPTPsZriEN+O0ovKXtkUXo0yQYpYV/qLp2Hqrq35zh/2\\neBWolxbu6kQ1doypjosGMAWhV2KpNTbglXYRdma1zZuI/mWH2sfVDuGUHXOszz3a\\noHLQfdjZFstSAsh1JgdY3iHo8uVPrBfwVpcXdX2xUW0s8Tj3X4GY5vhc8cysF/VA\\nDEAlhsxWqanZTslQrtGpuV3q0VoHaFVmf5XcxHm7IOo7UFbbmiOh2WLi22Uv7NxA\\nemqWEGzgU/j/aYoLA2YxmXmuebMTkAwuYfRYEJQ/m2/P/dnPUWRWKp6goaii+B1o\\n/SgfVxet9yM6ChUi7DVc4uRMjCDP/GkY6c26jsrLpQKBgQD4eCMFbXRb1JZGaH6D\\nIqrk+/To9wqxKOI1J3ti2mFLPG9+Pf4Erey4GkmXtFOjDxHfNq9Q4gNi2MDJDgYe\\nglnt4foPG3lqaus3cY7TcVv3cBRfMp1OX1v0Q5MmMyrcVKm9kLueFluLvSY8J8Bw\\nSF37YmWFmdANbxs2tybyTtQtLwKBgQDtIfwwlP6cSCgXaFzuQnkptLn0M5PQ8A0F\\nErxLwLnAgvIP5xI7AtOjVx0Uu7x5XGiTWnIhLU9ODkGQatcxocTmKOQ63oFiOEze\\n33XZKo1gq/ZDx7LGWpDmBR3xqgk/HwlzmuKlR0SqcS+xuU84mrNDqB8ayf6GCkao\\nYRnlE/WwbQKBgFxQXkqc8PdRU4fTOPXFwpKS3dpUNp+9ndW71obStgU67f2MUL0y\\nVVnNQnxfnhdd+Pjim15EqpdmCrJoSHO7YGgWZk6ImaKlGMEfqr36Rv32oUsBRhqh\\nKUvmc1xk9E6qEeqBRIOmsNqJKxR8fG37JRfJ5gguLnNfTVAV2h16ljA3AoGBAN+z\\nnNYz6JGEHJYgdPKrsOOgQ4BVG9ASdSXhC9Mmx9UNcs9/vBoBS6geqSeDB4UxoNHJ\\nlDsqJFNNbZqQv9tpcXdzAgNrHoGK/TGPevxYgTC+aL5+aG9oxqLIFvyA3OI4JFFz\\nvvYOan+j8Utmto5+mjhsJJPAFKVcklWL7MLHdpJtAoGAQHnyuT12Dzi3SeOMavhD\\nheJ6ant6pzH7bfifWcz558IfLtnxtYFR4azKW6n8SUnVCyybPMwVt4pxFZ6as0zK\\nKaplMuWadDqTSE56LJmG854fxQWq4/Z0qnYr9Yq7UHSRtNT7xOxyfiJkTVYqAT6q\\nYcPfY81+aw4wdjnWrDw5dSk=\\n-----END PRIVATE KEY-----\\n\",\n"
+            + "  \"client_email\": \"dummy@my-test-project.iam.gserviceaccount.com\",\n"
+            + "  \"client_id\": \"1111111111111111111\",\n"
+            + "  \"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\n"
+            + "  \"token_uri\": \"https://oauth2.googleapis.com/token\",\n"
+            + "  \"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\n"
+            + "  \"client_x509_cert_url\": \"https://www.googleapis.com/robot/v1/metadata/x509/dummy%40my-test-project.iam.gserviceaccount.com\"\n"
+            + "}";
+    String credentialsPath = writeToFile(credentials, "credentials.json");
+    setEnvironmentVariable("GOOGLE_APPLICATION_CREDENTIALS", credentialsPath);
```

Comment:
https://github.com/stefanbirkner/system-lambda is, to me, a cleaner way to set environment variables.  For example, add
```
  testImplementation "com.github.stefanbirkner:system-lambda:1.2.1"
```
and then (untested, but I think it's close):
```
SystemLambda.withEnvironmentVariable("GOOGLE_APPLICATION_CREDENTIALS", credentialsPath)
    .execute(() -> {
          assertDoesNotThrow(() -> SecretManagerServiceClient.create());
             });
```

Reply:
Ping on this @kirangodishala .  Are you OK if I make this change?

Reply:
I'm going ahead with this to keep making progress towards 1.29: https://github.com/spinnaker/clouddriver/pull/5773/commits/9def997f8640eace089979d7d19b1b7d69b16edc